Mollie/API/Object/Payment.py

```diff
@@ -2,22 +2,42 @@
 
 
 class Payment(Base):
-    STATUS_OPEN = 'open'
-    STATUS_PENDING = 'pending'
-    STATUS_CANCELLED = 'cancelled'
-    STATUS_EXPIRED = 'expired'
-    STATUS_PAID = 'paid'
-    STATUS_PAIDOUT = 'paidout'
-    STATUS_REFUNDED = 'refunded'
+    STATUS_OPEN         = 'open'
+    STATUS_PENDING      = 'pending'
+    STATUS_CANCELLED    = 'cancelled'
+    STATUS_EXPIRED      = 'expired'
+    STATUS_PAID         = 'paid'
+    STATUS_PAIDOUT      = 'paidout'
+    STATUS_REFUNDED     = 'refunded'
+    STATUS_FAILED       = 'failed'
+    STATUS_CHARGED_BACK = 'charged_back'
 
     def isOpen(self):
         return self['status'] == self.STATUS_OPEN
 
     def isPending(self):
         return self['status'] == self.STATUS_PENDING
 
+    def isCancelled(self):
+        return self['status'] == self.STATUS_CANCELLED
+
+    def isExpired(self):
+        return self['status'] == self.STATUS_EXPIRED
+
     def isPaid(self):
-        return 'paidDatetime' in self and self['paidDatetime']
```

Comment:
This is a tricky state. The line that has been removed is actually the correct behaviour. Checking for `isPaid()` should always succeed if there is a `paidDatetime`. This is because a payment with the state `paid` can be overridden by the state `paidout`. When a payment has the state `paidout`, `isPaid()` should still return true.